django/contrib/gis/gdal/raster/source.py

```diff
@@ -425,6 +425,25 @@ def warp(self, ds_input, resampling='NearestNeighbour', max_error=0.0):
 
         return target
 
+    def clone(self):
+        "Return a clone of this GDALRaster object"
+        return GDALRaster(
+            {
+                "width": self.width,
+                "height": self.height,
+                "srid": self.srid,
+                "origin": (self.origin.x, self.origin.y),
+                "scale": (self.scale.x, self.scale.y),
+                "skew": (self.skew.x, self.skew.y),
+                "bands": [{
+                    "data": band.data(), 
+                    "datatype": band.datatype()
+                } for band in self.bands],
+                **self.metadata
+            },
+            write=self._write
+        )
```

Comment:
Why this solution doesn't use `capi.copy_ds()`? :thinking:

Reply:
There is more than one way to instantiate a GDALRaster, this was experimental